### ISSUE TITLE
Historic Project Not Working

### ISSUE THREAD

**Issue by stephenhendry** [Azure/autorest#4529]
I have inherited a project that uses autorest. I have done a clean install on node (v16.15.0).

I have installed autorest using` npm install -g autorest`.

My projects come with an autorest.md file with the following:

```
# Acme API
> see https://aka.ms/autorest

``` yaml
version: 2.0.4238
input-file: ./swagger.json 
csharp: 
   namespace: Acme.API
   output-folder: ./AcmeAPI
   override-client-name: AcmeApiClient
   license-header: "This file was generated by AutoRest."
   add-credentials: true
```

My documentation says to run

 **autorest  .\autorestconf.md**

This returns

```
AutoRest code generation utility [cli version: 3.6.1; node: v16.15.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: 1.0.0001.
Failure:
Error: Unable to find a valid AutoRest core package '@autorest/core' @ '1.0.0001'.
Error: Unable to find a valid AutoRest core package '@autorest/core' @ '1.0.0001'.
    at selectVersion (C:\Users\Acme\AppData\Roaming\npm\node_modules\autorest\dist\src_autorest-as-a-service_ts.js:385:23)
    at resolveCoreVersion (C:\Users\Acme\AppData\Roaming\npm\node_modules\autorest\dist\src_autorest-as-a-service_ts.js:578:29)
    at main (C:\Users\Acme\AppData\Roaming\npm\node_modules\autorest\dist\app.js:98:33)
```

So I try running with the `--v3` flag

 **autorest --v3 .\autorestconf.md**

```
AutoRest code generation utility [cli version: 3.6.1; node: v16.15.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: ^3.2.0.
info    |    Loading AutoRest core      'C:\Users\Acme\.autorest\@autorestcore@3.8.4\nodemodules\@autorest\core\dist' (3.8.4)
info    |    Loading AutoRest extension '@autorest/csharp' (latest->3.0.0-beta.20220503.1)
info    |    Loading AutoRest extension '@autorest/modelerfour' (4.23.3->4.23.3)
warning | PreCheck/CheckDuplicateSchemas | Checking for duplicate schemas, this could take a (long) while.  Run with --verbose for more detail.
warning | ? | Operation '/AddUser' really should not have a media type (because there should be no body)
error   | DuplicateOperation | Duplicate Operation '' > 'TriggerUpdate' detected(This is most likely due to 2 operation using the same 'operationId' or 'tags'). Duplicates have those paths:
  - post /api/SP/TriggerUpdate
  - post /api/SP/TriggerNewUserUpdate
fatal   | Error: 1 errors occured -- cannot continue.
fatal   | Process() cancelled due to failure
error   |   Error: Plugin checker reported failure.
error   | Autorest completed with an error. If you think the error message is unclear, or is a bug, please declare an issues at https://github.com/Azure/autorest/issues with the error message you are seeing.
```
So I now try with the` --legacy` flag as well and get:

**autorest --v3 .\autorestconf.md --legacy**

```
AutoRest code generation utility [cli version: 3.6.1; node: v16.15.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: ^3.2.0.
info    |    Loading AutoRest core      'C:\Users\Acme\.autorest\@autorestcore@3.8.4\nodemodules\@autorest\core\dist' (3.8.4)
info    |    Loading AutoRest extension '@microsoft.azure/autorest.csharp' (~2.3.79->2.3.82)
info    |    Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.55->2.3.55)
fatal   | Process() cancelled due to failure
error   |   Error: Plugin 'commonmarker' not found.
error   | Autorest completed with an error. If you think the error message is unclear, or is a bug, please declare an issues at https://github.com/Azure/autorest/issues with the error message you are seeing.
```

Another post I saw said to use the flag `--use:@microsoft.azure/autorest.csharp@2.3.82`

**autorest --v3 .\autorestconf.md --use:@microsoft.azure/autorest.csharp@2.3.82**

```
AutoRest code generation utility [cli version: 3.6.1; node: v16.15.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
info    | AutoRest core version selected from configuration: ^3.2.0.
info    |    Loading AutoRest core      'C:\Users\Acme\.autorest\@autorestcore@3.8.4\nodemodules\@autorest\core\dist' (3.8.4)
info    |    Loading AutoRest extension '@microsoft.azure/autorest.csharp' (2.3.82->2.3.82)
info    |    Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.55->2.3.55)
fatal   | Process() cancelled due to failure
error   |   Error: Plugin 'commonmarker' not found.
error   | Autorest completed with an error. If you think the error message is unclear, or is a bug, please declare an issues at https://github.com/Azure/autorest/issues with the error message you are seeing.
```

Would you be able to give any pointers to get this code generating again? It did all work before when added to source code, the versions of node and autorest has changed since then.

**Comment by timotheeguerin** [comment]
Hey @stephenhendry could you try with just `--v2` (don't put --v3). Make sure the version of autorest core that is loaded is 2.x. That is the error you are getting with output 3&4. 

For output 2 does seems like something needs to be updated in the spec. Every operation needs unique ID

For the 1st one not sure what is going on here. Is there maybe more to your readme? Do you know where is `1.0.0001` coming from?

**Comment by stephenhendry** [comment]
@timotheeguerin  thanks for your reply.

`autorest --v2 .\autorestconf.md --legacy` gave me the same errors with `Error: Unable to find a valid AutoRest core package '@autorest/core' @ '1.0.0001'` in.

In the same folder I had other md files which had this in:

`version: 1.0.0001`

I removed `version: 1.0.0001 from all the md files and the `autorest --v2 .\autorestconf.md --legacy` now works! I also removed `version: 2.0.4238` from the file I was trying to use as that version could not be found. It defaults to version 2.0.4413.
